src/java/voldemort/utils/StoreDefinitionUtils.java

```diff
@@ -254,6 +255,102 @@ public static void validateSchemasAsNeeded(Collection<StoreDefinition> storeDefi
         }
     }
 
+    /**
+     * Ensure that new store definitions that are specified for an update do not include breaking changes to the store.
+     * @param oldStoreDefs
+     * @param newStoreDefs
+     */
+    public static void validateNewStoreDefsAreNonBreaking(List<StoreDefinition> oldStoreDefs, List<StoreDefinition> newStoreDefs){
+        Map<String, StoreDefinition> oldStoreMap = new HashMap<String, StoreDefinition>();
+        Map<String, StoreDefinition> newStoreMap = new HashMap<String, StoreDefinition>();
+        for (StoreDefinition storeDef : oldStoreDefs){
+           oldStoreMap.put(storeDef.getName(), storeDef);
+        }
+        for (StoreDefinition storeDef : newStoreDefs){
+           newStoreMap.put(storeDef.getName(), storeDef);
+        }
+        for (String storeName : oldStoreMap.keySet()){
+            if (newStoreMap.containsKey(storeName)){
+               validateNewStoreDefIsNonBreaking(oldStoreMap.get(storeName), newStoreMap.get(storeName));
+            }
+        }
+    }
+
+    /**
+     * Ensure that new store definitions that are specified for an update do not include breaking changes to the store.
+     *
+     * Non-breaking changes include changes to
+     *   description
+     *   preferredWrites
+     *   requiredWrites
+     *   preferredReads
+     *   requiredReads
+     *   retentionPeriodDays
+     *   retentionScanThrottleRate
+     *   retentionFrequencyDays
+     *   viewOf
+     *   zoneCountReads
+     *   zoneCountWrites
+     *   owners
+     *   memoryFootprintMB
+     *
+     * non breaking changes also include changes to the serializer type (name field) for
```

Comment:
Should it be called as Serializer definition can be changed, but not the Serilaizer Type itself.

Voldemort has inconsistent schema forward and backward compatibility checks.